Esri/arcgis-runtime-samples-dotnet
# Fix iOS memory issues by following viewcontroller lifecycle best practice

Previously, the iOS samples weren't using the viewcotnroller lifecycle methods to subscribe and unsubscribe from events. This caused a memory leak due to how the garbage collector is implemented on Xamarin.iOS. 

The memory leak issue would cause a crash if several samples were used in sequence, because iOS kills apps that use too much memory. 

Changes:

* `UIBarButtonItem`s are held in fields
* Event subscriptions added in `ViewWillAppear`
* Event subscriptions removed in `ViewDidDisappear`
* TableView references to the owning ViewController are marked `[Weak]`
* Jobs and timers are canceled in `ViewDidDisappear`
* Comments have been standardized in a few relevant areas, mainly by the UI field declarations, in `ViewWillAppear` and `ViewDidDisappear`, and in `LoadView`. 

How to test:

I've already tested all samples to confirm that the memory leak issues are fixed. This is a large change with relatively high risk of error, especially when managing event re-subscriptions (sample startup). 

To test, all samples should be opened and closed. If there are any leaks, the app will be killed before all samples can be run. 

Note: the following samples have lingering memory issues, primarily due to a difficult-to-debug issue with Xamarin.Auth:

* Author map
* Search a portal for maps
* Generate an offline map
* Generate an offline map with overrides
* Integrated windows authentication
* OAuth
* Token secured challenge
* Stats query group and sort

Those samples should be excluded from the test procedure defined above.
